mtcpclient/reporting.go

```diff
@@ -38,3 +38,45 @@ func StartBackgroundReporting(numberConnections int, rinterval int) (chan tcpcli
 
 	return connStatusCh, connStatusTracker
 }
+
+func FinalMetricsReport(connectionDescriptions []tcpclient.Connection) string {
+	mr := calculateMetricsReport(connectionDescriptions)
+	return "Time to establish TCP connections min/avg/max/stdDev = " +
+		mr.minToEstablished.String() + "/" +
+		mr.avgToEstablished.String() + "/" +
+		mr.maxToEstablished.String() + "/" +
+		mr.stdDevToEstablished.String()
+}
+
+type metricsReport struct {
+	avgToEstablished    time.Duration
+	minToEstablished    time.Duration
+	maxToEstablished    time.Duration
+	totalToEstablished  time.Duration
+	stdDevToEstablished time.Duration
+}
+
+func calculateMetricsReport(c []tcpclient.Connection) metricsReport {
```

Comment:
The function an struct maybe can be moved to a different go file (metricsCalcs.go?), as you have nicely decoupled this from the report. But, there's a function doing something similar to this (PrintGroupOfConnections) as, given a group of connections, "it summarizes them" in some way. And I decided to put that close to the connection type definition (despite its an operation over a group of connections), in the tcpclient package. I'm not saying that was better than this, but maybe we need to align them... maybe its type to create our custom type for a "group of connections", being these functions just "methods" of this new type? My original PrintGroupOfConnections function could be the standard "String()" method, while this one would be a "consolidateMetrics" of a connection group...

Reply:
mmm, it makes sense... 😉